libresourceqt/src/resource-engine.cpp

```diff
@@ -660,7 +724,11 @@ bool ResourceEngine::registerVideoProperties(quint32 pid)
 static void connectionIsUp(resconn_t *connection)
 {
     qCDebug(lcResourceQt, "**************** %s() - locking....", __FUNCTION__);
+#if (QT_VERSION < QT_VERSION_CHECK(6,0,0))
     QMutexLocker locker(&mutex);
+#else
+    mutex.lock();
```

Comment:
Not following why this change? Different behavior too, nothing releases the mutex when the function is exited? Doesn't feel proper.

Reply:
Was resolved without explanation, unresolving.

QRecursiveMutex is Qt 5.14 so using that would be better, but I'm not following why a recursive mutex would be needed because of Qt version.

And then QMutexLocker works with a recursive mutex too so I'm not following why these explicit locks which are even missing the unlocks.